client/driver/spawn/spawn_test.go

```diff
@@ -5,64 +5,81 @@ import (
 	"io/ioutil"
 	"os"
 	"os/exec"
-	"runtime"
 	"strings"
 	"testing"
 	"time"
 )
 
-func TestSpawn_NoCmd(t *testing.T) {
-	f, err := ioutil.TempFile("", "")
-	if err != nil {
-		t.Fatalf("TempFile() failed")
+func TestMain(m *testing.M) {
+	switch os.Getenv("TEST_MAIN") {
+	case "app":
+		appMain()
+	default:
+		os.Exit(m.Run())
+	}
+}
+
+func appMain() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "no command provided")
+		os.Exit(1)
+	}
+	switch cmd := os.Args[1]; cmd {
+	case "echo":
+		fmt.Println(strings.Join(os.Args[2:], " "))
+	case "sleep":
+		if len(os.Args) != 3 {
+			fmt.Fprintln(os.Stderr, "expected 3 args")
+			os.Exit(1)
+		}
+		dur, err := time.ParseDuration(os.Args[2])
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "could not parse sleep time: %v", err)
+			os.Exit(1)
+		}
+		time.Sleep(dur)
+	default:
+		fmt.Fprintln(os.Stderr, "unknown command:", cmd)
+		os.Exit(1)
 	}
-	defer os.Remove(f.Name())
+}
 
-	spawn := NewSpawner(f.Name())
+func TestSpawn_NoCmd(t *testing.T) {
+	t.Parallel()
+	tempFile := tempFileName(t)
+	defer os.Remove(tempFile)
+
+	spawn := NewSpawner(tempFile)
 	if err := spawn.Spawn(nil); err == nil {
 		t.Fatalf("Spawn() with no user command should fail")
 	}
 }
 
 func TestSpawn_InvalidCmd(t *testing.T) {
-	f, err := ioutil.TempFile("", "")
-	if err != nil {
-		t.Fatalf("TempFile() failed")
-	}
-	defer os.Remove(f.Name())
+	t.Parallel()
+	tempFile := tempFileName(t)
+	defer os.Remove(tempFile)
 
-	spawn := NewSpawner(f.Name())
-	spawn.SetCommand(exec.Command("foo"))
+	spawn := NewSpawner(tempFile)
+	spawn.SetCommand(exec.Command("foo")) // non-existent command
 	if err := spawn.Spawn(nil); err == nil {
-		t.Fatalf("Spawn() with no invalid command should fail")
+		t.Fatalf("Spawn() with an invalid command should fail")
 	}
 }
 
 func TestSpawn_SetsLogs(t *testing.T) {
-	// TODO: Figure out why this test fails. If the spawn-daemon directly writes
```

Comment:
I am not sure. I just checked out master at a point before these changes and removed the call to `t.Skip`. With that code `go test` fails on my machine because "echo" isn't available on the path.

```
C:\nomadproject\src\github.com\hashicorp\nomad\client\driver\spawn>git describe
v0.1.2-615-g9db262f

C:\nomadproject\src\github.com\hashicorp\nomad\client\driver\spawn>go test -v -run SetsLogs
=== RUN   TestSpawn_SetsLogs
--- FAIL: TestSpawn_SetsLogs (0.03s)
        spawn_test.go:60: Spawn() failed: Failed to execute user command: Error starting user command: exec: "echo": executable file not found in %PATH%
FAIL
exit status 1
FAIL    github.com/hashicorp/nomad/client/driver/spawn  0.118s
```

So I would say the fix was adding `TestMain` and using the binary constructed by `go test` as a self packaged portable `echo`.

But the TODO comment seems to indicate that you had experienced a different type of failure where the `echo` command could be found—maybe `go test` was running in a Cygwin or msys shell—but couldn't write to the log file for some reason. So the problem doesn't seem to reproduce the same way for me.